front/lib/api/gens.ts

```diff
@@ -69,9 +80,14 @@ export async function getTemplate(
 }
 
 export async function updateTemplate(
-  template: GensTemplateType,
-  owner: WorkspaceType
+  auth: Authenticator,
+  template: GensTemplateType
 ) {
+  const owner = auth.workspace();
+  if (!owner) {
+    throw new Error("Unexpected `auth` without `workspace`.");
```

Comment:
feels a bit weird to consider !owner should throw for update, but return kindly for get

I guess the logic is "we're f*cked" for updates, and "we can recover" for get? Since not sure, asking

Reply:
Yes exactly

Reply:
This can be revisited but this is aligned with what we do in other files in this directory

Reply:
Technically we could and probably should recheck here that we are authorized to write with the current Authenticator